d2/src/main/java/com/linkedin/d2/balancer/dualread/DualReadLoadBalancer.java

```diff
@@ -104,12 +105,17 @@ public void start(Callback<None> callback)
 
     // if in new-lb-only mode, new lb needs to start successfully to call the callback. Otherwise, the old lb does.
     // Use a separate executor service to start the new lb, so both lbs can start concurrently.
-    if (!_newLbExecutor.isShutdown())
+    try
     {
       _newLbExecutor.execute(() -> _newLb.start(getStartUpCallback(true,
               mode == DualReadModeProvider.DualReadMode.NEW_LB_ONLY ? callback : null)
       ));
     }
+    catch (RejectedExecutionException e)
+    {
+      _rateLimitedLogger.debug("newLb executor rejected new task for start. "
+          + "It is shut down or its queue size has reached max limit");
+    }
```

Comment:
Should this retry at all? Or is it OK to not warm up here? Maybe leave a comment to that effect, just in case

Reply:
If it's already being shutdown at startup, it's very edge case, and should shut down without retry. 
It won't hit the queue size at startup for sure, so don't need to worry about that case.